stats/opentelemetry/csm/pluginoption.go

```diff
@@ -311,26 +298,15 @@ func getNodeID() string {
 // metadataExchangeKey is the key for HTTP metadata exchange.
 const metadataExchangeKey = "x-envoy-peer-metadata"
 
-func determineTargetCSM(target string) bool {
+func determineTargetCSM(parsedTarget *url.URL) bool {
 	// On the client-side, the channel target is used to determine if a channel is a
 	// CSM channel or not. CSM channels need to have an “xds” scheme and a
 	// "traffic-director-global.xds.googleapis.com" authority. In the cases where no
 	// authority is mentioned, the authority is assumed to be CSM. MetadataExchange
 	// is performed only for CSM channels. Non-metadata exchange labels are detected
 	// as described below.
-	parsedTarget, err := url.Parse(target)
-	if err != nil {
-		// Shouldn't happen as Dial would fail if target couldn't be parsed, but
-		// log just in case to inform user.
-		logger.Errorf("passed in target %v failed to parse: %v", parsedTarget, err)
-		return false
-	}
 
-	if parsedTarget.Scheme == "xds" {
-		if parsedTarget.Host == "" {
-			return true // "In the cases where no authority is mentioned, the authority is assumed to be csm"
-		}
-		return parsedTarget.Host == "traffic-director-global.xds.googleapis.com"
-	}
-	return false
+	// "In the cases where no authority is mentioned, the authority is assumed
+	// to be csm".
```

Comment:
This is already above?

Reply:
Oh whoops sorry this got left behind when the top level comment was far because I had a lot more lines before this single line.